File: module-wanandroid/src/main/java/com/yyxnb/module_wanandroid/viewmodel/WanSearchViewModel.java
package com.yyxnb.module_wanandroid.viewmodel;

import androidx.lifecycle.MutableLiveData;

import com.yyxnb.common_base.base.CommonViewModel;
import com.yyxnb.common_res.bean.WanData;
import com.yyxnb.common_res.config.Http;
import com.yyxnb.module_wanandroid.bean.WanAriticleBean;
import com.yyxnb.module_wanandroid.bean.WanClassifyBean;
import com.yyxnb.module_wanandroid.bean.WanStatus;
import com.yyxnb.module_wanandroid.config.WanService;

import java.util.List;

public class WanSearchViewModel extends CommonViewModel {

    private final WanService mApi = Http.getInstance().create(WanService.class);

    public MutableLiveData<List<WanClassifyBean>> searchData = new MutableLiveData<>();
    public MutableLiveData<WanStatus<WanAriticleBean>> searchDataByKey = new MutableLiveData<>();

    public void getSearchData() {

        launchOnlyResult(mApi.getSearchData(), new HttpResponseCallback<WanData<List<WanClassifyBean>>>() {
            @Override
            public void onSuccess(WanData<List<WanClassifyBean>> data) {
                searchData.postValue(data.getResult());
            }

            @Override
            public void onError(String msg) {
            }
        });
    }

    public void getSearchDataByKey(int page, String key) {
        launchOnlyResult(mApi.getSearchDataByKey(page, key), new HttpResponseCallback<WanData<WanStatus<WanAriticleBean>>>() {
            @Override
            public void onSuccess(WanData<WanStatus<WanAriticleBean>> data) {
                searchDataByKey.postValue(data.getResult());
            }

            @Override
            public void onError(String msg) {
            }
        });
    }

}

File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/viewmodel/WanSearchViewModel.kt
package com.yyxnb.module_wanandroid.viewmodel

import androidx.lifecycle.MutableLiveData
import com.yyxnb.common.utils.log.LogUtils
import com.yyxnb.common_base.config.Http
import com.yyxnb.module_wanandroid.bean.WanAriticleBean
import com.yyxnb.module_wanandroid.bean.WanClassifyBean
import com.yyxnb.module_wanandroid.bean.WanData
import com.yyxnb.module_wanandroid.bean.WanStatus
import com.yyxnb.module_wanandroid.config.WanService
import com.yyxnb.network.BaseViewModel

class WanSearchViewModel : BaseViewModel() {

    private val mApi = Http.create(WanService::class.java)

    @JvmField
    val searchData = MutableLiveData<List<WanClassifyBean>>()

    @JvmField
    val searchDataByKey = MutableLiveData<WanStatus<WanAriticleBean>>()

    fun getSearchData() {
        launchOnlyResult(
                block = { mApi.searchData() },
                reTry = { getSearchData() },
                success = { searchData.postValue(it) },
                error = { LogUtils.e(it.message) }
        )
    }

    fun getSearchDataByKey(page: Int, key: String?) {
        launchOnlyResult(
                block = { mApi.getSearchDataByKey(page, key) },
                reTry = { getSearchDataByKey(page, key) },
                success = { searchDataByKey.postValue(it) },
                error = { LogUtils.e(it.message) }
        )
    }
}